source/common/http/async_client_impl.h

```diff
@@ -93,6 +93,7 @@ class AsyncStreamImpl : public AsyncClient::Stream,
 public:
   AsyncStreamImpl(AsyncClientImpl& parent, AsyncClient::StreamCallbacks& callbacks,
                   const AsyncClient::StreamOptions& options);
+  ~AsyncStreamImpl() override { router_.onDestroy(); }
```

Comment:
Sorry for the late drive-by here, but quick question: would it make sense for `Router::ProdFilter` or `Router::Filter`'s destructor to call its own `onDestroy`? (I'm looking at the comments on `StreamFilterBase::onDestroy` but I don't really understand whether they're applicable here.)